packages/stencil-component/src/utils/utils.ts

```diff
@@ -64,72 +67,96 @@ function getNodeByXpath(xpath: string, xml: Document): Node[] {
  */
 export function zip(arrays): Array<any[]> {
   return arrays[0].map(function (_, i) {
-    return arrays.map(function (array) { return array[i] })
+    return arrays.map(function (array) {
+      return array[i]
+    })
   });
 }
 
 
 /**
  * Return sentences from TEI xml file
- * @param {string} - the path to the TEI file
+ * @param {string} path the path to the TEI file
  */
-export function parseTEI(path: string): Page[] {
-  let xmlDocument =  getXML(path)
+export async function parseTEI(path: string): Promise<Page[]> {
+  let xmlDocument
+  try {
+    xmlDocument = await fetchDoc(path)
+  } catch (e) {
+
+    return Promise.reject(e)
+  }
   let parser = new DOMParser();
   let xml_text = parser.parseFromString(xmlDocument, "text/xml")
-  let pages = getNodeByXpath('.//div[@type="page"]', xml_text)
-  let parsed_pages = pages.map((p: Element) => {
-    let id = p.id;
-    let img_xpath = `.//div[@id='${id}']/graphic/@url`
-    let img = getNodeByXpath(img_xpath, xml_text)
-    let p_xpath = `.//div[@id='${id}']/p`
-    let paragraphs = getNodeByXpath(p_xpath, xml_text)
-    let parsed_page = { id: id, paragraphs: paragraphs }
-    if (img.length > 0) {
-      parsed_page['img'] = img[0].nodeValue;
-    }
-    if(p.attributes)parsed_page["attributes"]=p.attributes;
-    return parsed_page
-  });
-  return parsed_pages
-}
+  try {
+    let pages = getNodeByXpath('.//div[@type="page"]', xml_text)
+    return pages.map((p: Element) => {
+      let id = p.id;
+      let img_xpath = `.//div[@id='${id}']/graphic/@url`
+      let img = getNodeByXpath(img_xpath, xml_text)
+      let p_xpath = `.//div[@id='${id}']/p`
+      let paragraphs = getNodeByXpath(p_xpath, xml_text)
+      let parsed_page = {id: id, paragraphs: paragraphs}
+      if (img.length > 0) {
+        parsed_page['img'] = img[0].nodeValue;
+      }
+      if (p.attributes) parsed_page["attributes"] = p.attributes;
+      return parsed_page
+    });
 
+  } catch (e) {
```

Comment:
Better error handling is appreciated!  The parsing code changed entirely in dev.readalong-format though, to eliminate the use of XPath and associated namespace-related confusion